azure/durable_functions/tasks/signal_entity.py

```diff
@@ -0,0 +1,52 @@
+from typing import List, Any, Optional
+
+from ..models.Task import (
+    Task)
+from ..models.actions.SignalEntityAction import SignalEntityAction
+from ..models.history import HistoryEvent, HistoryEventType
+from .task_utilities import set_processed, parse_history_event, find_event
+from ..models.utils.entity_utils import EntityId
+from ..models.entities.RequestMessage import RequestMessage
+from ..models.entities.ResponseMessage import ResponseMessage
+import logging
+
+
+def signal_entity_task(
+        context,
+        state: List[HistoryEvent],
+        entity_id: EntityId,
+        operation_name: str = "",
+        input_: Optional[Any] = None):
+    """ Signal a entity operation.
+
+    It the action hasn't been scheduled, it appends the action. If the action has been scheduled, no ops.
+
+    Parameters
+    ----------
+    state: List[HistoryEvent]
+        The list of history events to search over to determine the
+        current state of the callEntity Task.
+    entity_id: EntityId
+        An identifier for the entity to call.
+    operation_name: str
+        The name of the operation the entity needs to execute.
+    input_: Any
+        The JSON-serializable input to pass to the activity function.
+
+    """
+    new_action = SignalEntityAction(entity_id, operation_name, input_)
+    scheduler_id = EntityId.get_scheduler_id(entity_id=entity_id)
+
+    hist_type = HistoryEventType.EVENT_SENT
+    extra_constraints = {
+        "InstanceId": scheduler_id,
+        "Name": "op"
+    }
+
+    event_sent = find_event(state, hist_type, extra_constraints)
+    set_processed([event_sent])
+
+    if event_sent:
+        return 
+
+    context.actions.append([new_action])
```

Comment:
Since `signal_entity` is _weird_ in that it does not wait for a response, I think this portion would benefit from just a few comments explaining what's going on and why